### PR TITLE
Daniel Briechle BE Mod 2 

### PR DESCRIPTION
**Pull request by danbriechle**

### CODE REVIEW

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 1]
```diff
@@ -52,11 +54,11 @@
 
   it '2. finds order id of smallest order' do
     # ----------------------- Using Raw SQL ----------------------
-    order_id = ActiveRecord::Base.connection.execute('SELECT id FROM orders ORDER BY amount ASC LIMIT 1').first['id']
+    # order_id = ActiveRecord::Base.connection.execute('SELECT id FROM orders ORDER BY amount ASC LIMIT 1').first['id']
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    order_id = Order.first[:id]
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: Coincidentally, this was the "first" order in the database, but how else could you locate the ID of an order that has the smallest 'amount' ?

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 2]
```diff
@@ -146,11 +149,11 @@
     expected_result = [order_1, order_2, order_3, order_4]
 
     # ----------------------- Using Ruby -------------------------
-    orders_less_than_550 = Order.all.select { |order| order.amount < 550 }
+    # orders_less_than_550 = Order.all.select { |order| order.amount < 550 }
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    orders_less_than_550 = Order.where(amount: [0..549])
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: Technically, amounts less than 0 would also need to be found here, to `.where("amount < 550")` is best... (what if an order was a refund?)

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 3]
```diff
@@ -338,20 +342,21 @@
...
+    # Order.all.each do |order|
+    #   if order.items
+    #     grouped_orders << order if order.user_id == 3
+    #   end
+    # end
+    # grouped_orders.each_with_index do |order, idx|
+    #   items_for_user_3_third_order = order.items.map(&:name) if idx == 2
+    # end
+    # ------------------------------------------------------------
+    items_for_user_3_third_order = User.find(user_3.id).orders.find(9).items.pluck(:name)
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: If we knew the order ID was 9, `Order.find(9)` would solve this ... how else could you find this if you didn't know the ID of the order?

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 4]
```diff
@@ -442,13 +447,13 @@
...
-    order_ids = OrderItem.where(order_id: orders, item_id: item_4.id).map(&:order_id)
-    orders = order_ids.map { |id| Order.find(id) }
+    # orders = Order.where(user_id: user_2)
+    # order_ids = OrderItem.where(order_id: orders, item_id: item_4.id).map(&:order_id)
+    # orders = order_ids.map { |id| Order.find(id) }
     # -----------------------------------------------------------
 
     # ------------------ Improved Solution ----------------------
-    #  Solution goes here
+    orders = User.find(user_2.id).orders.joins(:order_items).where(order_items: {item: item_4})
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: different approach: Order.joins(order_items: :items).where(user_id: 2, order_items: {item: item_4})
starting on the User model would expect that the resulting data would be put back into User objects

**Comment by iandouglas** [comment]
@danbriechle there are a few comments I've left for you in here, can you re-visit those and take another try at them?

File: spec/models/activerecord_obstacle_course_spec.rb (hunk 4 above)
Comment (danbriechle) [reply spec/models/activerecord_obstacle_course_spec.rb]: its actually Order.joins(:order_items).where(user_id: 2, order_items: {item: item_4})

**Pull request closed without merging**